src/extractor/intersection/intersection_analysis.cpp

```diff
@@ -0,0 +1,808 @@
+#include "extractor/intersection/intersection_analysis.hpp"
+
+#include "util/assert.hpp"
+#include "util/bearing.hpp"
+#include "util/coordinate_calculation.hpp"
+
+#include "extractor/guidance/coordinate_extractor.hpp"
+
+#include <boost/optional/optional_io.hpp>
+
+namespace osrm
+{
+namespace extractor
+{
+namespace intersection
+{
+
+IntersectionEdges getIncomingEdges(const util::NodeBasedDynamicGraph &graph,
+                                   const NodeID intersection_node)
+{
+    IntersectionEdges result;
+
+    for (const auto outgoing_edge : graph.GetAdjacentEdgeRange(intersection_node))
+    {
+        const auto from_node = graph.GetTarget(outgoing_edge);
+        const auto incoming_edge = graph.FindEdge(from_node, intersection_node);
+
+        if (!graph.GetEdgeData(incoming_edge).reversed)
+        {
+            result.push_back({from_node, incoming_edge});
+        }
+    }
+
+    // Enforce ordering of incoming edges
+    std::sort(result.begin(), result.end());
+    return result;
+}
+
+IntersectionEdges getOutgoingEdges(const util::NodeBasedDynamicGraph &graph,
+                                   const NodeID intersection_node)
+{
+    IntersectionEdges result;
+
+    for (const auto outgoing_edge : graph.GetAdjacentEdgeRange(intersection_node))
+    {
+        // TODO: to use TurnAnalysis all outgoing edges are required, to be uncommented later
```

Comment:
I don't quite understand yet how we could skip `reversed` edges in the future. How will this be possible?